src/client/mod.rs

```diff
@@ -311,6 +336,42 @@ impl Message {
         }
     }
 
+    /// Gets the message's expected section PublicKey.
+    pub fn target_section_pk(&self) -> Option<PublicKey> {
+        match self {
```

Comment:
I think you can match them all together in a single arm just like we do in `id` function above?